### DISCUSSION TITLE
PID Macro

### DISCUSSION THREAD

**Discussion by Identity-Unkn0wn** [jschuh/klipper-macros#117] [Ideas]
Any thought on adding these jschuh? Not necessary, but I really like not having to remember the pid stuff and this also does the whole save_config afterwards as well. If this isn't the place for this I apologize, just really like your suite of macros.

```
[gcode_macro PID_BED]
description: PID Tune for the Bed
gcode:
  {% set TARGET_TEMP = params.TARGET_TEMP|default(60)|float %}
  PID_CALIBRATE HEATER=heater_bed TARGET={TARGET_TEMP} 
  TURN_OFF_HEATERS
  SAVE_CONFIG

[gcode_macro PID_EXTRUDER]
description: PID Tune for the Extruder gcode:
gcode:
  {% set TARGET_TEMP = params.TARGET_TEMP|default(200)|float %}
  PID_CALIBRATE HEATER=extruder TARGET={TARGET_TEMP}
  TURN_OFF_HEATERS 
  SAVE_CONFIG
```

**Comment by jschuh** [comment]
This is certainly the right place to ask the question. Although, I don't see a good argument for doing it, since it really just replaces two commands with one command.

**Reply by Identity-Unkn0wn** [reply]
Mostly just gives it a macro button that you could just enter the temp you want to PID and send it. Just an idea, just started learning macros and these are the first set i got working 😂😂

**Reply by jschuh** [reply]
Ah, then I think that sort of targeted UX personalization is a perfect reason to write your own custom macros. Whereas, I try to scope this project to addressing more complex functionality that fits a broad range of printer setups and use cases.
